src/lib/crypto/kimchi_backend/common/tests/test_lookup_table_constraint_kind.ml

```diff
@@ -37,15 +39,391 @@ let test_finalize_and_get_gates_with_lookup_tables () =
 let test_finalize_and_get_gates_with_runtime_table_cfg () =
   let module Tick = Kimchi_backend.Pasta.Vesta_based_plonk in
   let cs = Tick.R1CS_constraint_system.create () in
+
+  let indexed_runtime_table_cfg =
+    [| Tick.Field.zero
+     ; Tick.Field.one
+     ; Tick.Field.(one + one)
+     ; Tick.Field.(one + one + one)
+    |]
+  in
+
   let () =
     Tick.R1CS_constraint_system.(
-      add_constraint cs (T (AddRuntimeTableCfg { id = 1l; first_column = [||] })))
+      add_constraint cs
+        (T
+           (AddRuntimeTableCfg
+              { id = 1l; first_column = indexed_runtime_table_cfg } ) ))
   in
   let () = Tick.R1CS_constraint_system.set_primary_input_size cs 1 in
+  let _aux = Tick.R1CS_constraint_system.set_auxiliary_input_size cs 1 in
   let _gates, _lt, rt = Tick.R1CS_constraint_system.finalize_and_get_gates cs in
   assert (rt.(0).id = 1l) ;
   assert (Array.length rt = 1)
 
+let test_compute_witness_with_lookup_to_the_same_idx_twice () =
+  (* See the comment in compute_witness when populating the runtime tables. The
+     function does not check that the runtime table has already been set at a
+     certain position, and it overwrites the previously set value *)
+  let module Tick = Kimchi_backend.Pasta.Vesta_based_plonk in
+  let module Impl = Snarky_backendless.Snark.Run.Make (Tick) in
+  let table_id = 0 in
+  let table_size = 10 in
+  let table_id_var, idx1_var, v1_var, idx2_var, v2_var, idx3_var, v3_var =
+    Snarky_backendless.Cvar.(Var 0, Var 1, Var 2, Var 3, Var 4, Var 5, Var 6)
```

Comment:
This is super low-level. Probably better to define a circuit and use snarky APIs

Reply:
The idea of the test is to be low-level and try to not use Snarky as much as possible. The test is located in a subdirectory of common, therefore it is supposed to test features and use code from this library as much as possible.
I agree the code can be in somehow more high-level, I give a shot. However, calling `compute_witness` from `Plonk_constraint_system` is a requirement as the file is supposed to contain unit-test for this function.

Reply:
See a solution in aab87b9ee4292906a3f9e1e1899602c91ec5761c. I will duplicate for the others tomorrow.